nomad/structs/config/artifact.go

```diff
@@ -166,6 +186,22 @@ func (a *ArtifactConfig) Validate() error {
 		return fmt.Errorf("s3_timeout must be > 0")
 	}
 
+	if a.DecompressionFileCountLimit == nil {
```

Comment:
Indeed it "can't" ever be `nil` because we only create one of these from a `DefaultConfig()` helper; the validation here should not set a default at all I think, and simply fail the validation if the value is nil. 

Good catch!